### PR TITLE
General fixes and unit testing

### PR DESCRIPTION
**Pull request by artagel**

### CODE REVIEW

**Comment by artagel** [comment]
I just realized it wouldn't hurt to implement this in the official repo.  I went through and found all the places errors weren't handled, etc, and fixed them.

File: handlers/handlers.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func init() {
 func loginURL(r *http.Request, state string) string {
 	// State can be some kind of random generated hash string.
 	// See relevant RFC: http://tools.ietf.org/html/rfc6749#section-10.12
-	var url = ""
+	var theUrl = ""
```
Comment (bnfinet) [inline handlers/handlers.go]: why "theUrl"?

File: main.go [hunk 2]
```diff
@@ -55,23 +55,23 @@ func main() {
 		"listen", listen,
 		"oauth.provider", cfg.GenOAuth.Provider)
 
-	mux := mux.NewRouter()
+	theMux := mux.NewRouter()
```
Comment (bnfinet) [inline main.go]: why `theMux`?

**Comment by bnfinet** [comment]
thanks for these, adding the additional error checks is really great.

I'm not so into the "the" var name changes but everything else is good with the one exception of the `lasso` comment.  If you can rebase, make these fixes and then I'll merge.

Thanks @artagel !

File: handlers/handlers.go (hunk 1 above)
Comment (artagel) [reply handlers/handlers.go]: My IDE  was telling me those were names overlapping with some builtin stuff. So it was the simplest way to change them.  They can be anything but what they were.

File: main.go (hunk 2 above)
Comment (artagel) [reply main.go]: My IDE  was telling me those were names overlapping with some builtin stuff. So it was the simplest way to change them.  They can be anything but what they were.

**Pull request merged**